wandb/sdk/launch/runner/kubernetes.py

```diff
@@ -0,0 +1,222 @@
+import datetime
+import os
+import subprocess
+import time
+from typing import Any, Dict, List, Optional
+
+if False:
+    import kubernetes   # type: ignore
+from six.moves import shlex_quote
+import wandb
+import wandb.docker as docker
+from wandb.errors import LaunchError
+from wandb.util import get_module, load_json_yaml_dict
+import yaml
+
+from .abstract import AbstractRun, AbstractRunner, Status
+from .._project_spec import LaunchProject
+from ..docker import (
+    construct_local_image_uri,
+    generate_docker_image,
+    pull_docker_image,
+    validate_docker_installation,
+)
+from ..utils import (
+    PROJECT_DOCKER_ARGS,
+    PROJECT_SYNCHRONOUS,
+)
+
+TIMEOUT = 5
+
+class KubernetesSubmittedRun(AbstractRun):
+    from kubernetes.client.api.batch_v1_api import BatchV1Api
+
+    def __init__(self, batch_api: BatchV1Api, name: str, namespace: Optional[str] = 'default') -> None:
+        self.batch_api = batch_api
+        self.name = name
+        self.namespace = namespace
+        self.job = self.batch_api.read_namespaced_job(name=self.name, namespace=self.namespace)
+
+    @property
+    def id(self) -> str:
+        return self.name
+
+    def wait(self) -> bool:
+        while True:
+            status = self.get_status()
+            wandb.termlog("Job {} status: {}".format(self.name, status))
+            if status != "running":
+                break
+            time.sleep(5)
+        return status == "finished"     # todo: not sure if this (copied from aws runner) is the right approach? should we return false on failure
+
+    def get_status(self) -> Status:
+        job_response = self.batch_api.read_namespaced_job_status(name=self.name, namespace=self.namespace)
+        status = job_response.status
+        # todo: we only handle the 1 pod case. see https://kubernetes.io/docs/concepts/workloads/controllers/job/#parallel-jobs for multipod handling
+        if status.succeeded == 1:
+            return Status("finished")
+        elif status.failed == 1:
```

Comment:
nit: should use >=1 for failed, ive noticed k8s will retry your pod sometimes and end up with more than 1 failed containers.